### DISCUSSION TITLE
Welcome to Advanture-Game Discussions!

### DISCUSSION THREAD

**Discussion by eisenimmortal** [eisenimmortal/Advanture-Game#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by eisenimmortal** [comment]
Привет

**Comment by Rasatabula** [comment]
Hello world!

**Reply by eisenimmortal** [reply]
Пример Мокинга:

import unittest                                                        # Импортируется библиотека для тестирования
from unittest.mock import Mock                            # Импортируется модуль для мокинга


class A:                                                                    # Создаётся класс
    def __init__(self):                                                  # Эта штука говорит о том какие переменные будут в экземпляре класса (когда из 
                                                                                # класса будут создаваться объекты) 

        self.data_base = {"сил осталось": 5}              # Переменная, где показано до скольки восстановить 
        self.now = {"сил осталось": 0}                       # Переменная, где показано что восстановить


def restore_power(a):                                             # Определение функции "восстановить силы" - restore power
   a.now = a.data_base                                           # Что восстановить приравнивается на что восстановить
   return a.data_base                                              # И return то что на что восстановить (return таки появляется)


class TestA(unittest.TestCase):                              # Создаётся класс Теста, который наследует все функции из модуля TestCase, который 
                                                                             # находится в библиотеке  тестирования "unittest"
    def test_a(self):                                                 # Создаётся функция которая будет в экземпляре класса 
        mockobj = Mock(now=4, data_base=5)      # В ней создаётся экземпляр из импорт-го нами во второй строчке класса Mock
                                                                             # Если вдруг интересно что за класс Mock, то можно посмотреть оффициальную 
                                                                             # документацию https://docs.python.org/3/library/unittest.mock.html
        restore_power(mockobj)                              # в  функцию restore power передаётся аргумент mockobj (определён строкой выше)
        self.assertEqual(mockobj.now, mockobj.data_base)       # Происходит проверка mockobj.now  равен ли mockobj.data_base
                                                                             # Ну и соответственно функция вернёт True, если да и  False,  если нет

# По итогу, что нам дал мокинг: он нам позволил  обезличить класс (Class A) и вставить  в обезличеный класс необходимые параметры и проверить верные ли будут параметры на выходе. Как бы функция test_a выглядела без мокинка:

def test_a(self):                                                 
        not_mocking= A (now=4, data_base=5)          # "A" это класс, который мы определили наверху
        restore_power(not_mocking)                            # Дальше также как было
        self.assertEqual(not_mocking.now, not_mocking.data_base)
